File: web/Magazzino/Models/TipoUdcModel.cs
using System.Data.SqlClient;
using web.Generale;

namespace web.Magazzino.Models
{
    public class TipoUdcModel : BasicModel
    {
        public int Id_tipo_udc { get; set; }

        public string? Cod_tipo_udc { get; set; }

        public string? Desc_tipo_udc { get; set; }

        public bool Uscita { get; set; }


        public TipoUdcModel()
        {
            NomeTabella = "tipo_udc";
        }



        public override void FromReader(SqlDataReader reader)
        {
            FromReaderCommon(reader);

            Id_tipo_udc = GetIntNonNullable(reader, nameof(Id_tipo_udc));
            Cod_tipo_udc = GetString(reader, nameof(Cod_tipo_udc));
            Desc_tipo_udc = GetString(reader, nameof(Desc_tipo_udc));
            Uscita = GetBoolean(reader, nameof(Uscita))??false;
        }
    }
}

File: app/lib/magazzino/models/tipo_udc_model.dart
// ignore_for_file: non_constant_identifier_names

import 'package:ows_test_v2/generale/models/abstract_model.dart';

class TipoUdcModel extends AbstractModel {
  int? id_tipo_udc;
  String? cod_tipo_udc;
  String? desc_tipo_udc;
  bool? uscita;

  TipoUdcModel({
    this.id_tipo_udc,
    this.cod_tipo_udc,
    this.desc_tipo_udc,
    this.uscita,
  });

  factory TipoUdcModel.fromJson(Map<String, dynamic> json) {
    return TipoUdcModel(
      id_tipo_udc: json['id_tipo_udc'],
      cod_tipo_udc: json['cod_tipo_udc'],
      desc_tipo_udc: json['desc_tipo_udc'],
      uscita: json['uscita'],
    );
  }

  @override
  Map<String, dynamic> toJson() {
    return <String, dynamic>{
      'id_tipo_udc': id_tipo_udc,
      'cod_tipo_udc': cod_tipo_udc,
      'desc_tipo_udc': desc_tipo_udc,
      'uscita': uscita,
    };
  }
}
